server/src/main.rs

```diff
@@ -67,30 +67,49 @@ fn main() -> Result<()> {
 
 /// Thread which listens for new connections and sends them to the given MPSC channel
 /// Technically we could use a non-blocking connection accepter, but it was easier not to for now
-fn connection_listener(addr: SocketAddr, conn_tx: Sender<(TcpStream, SocketAddr)>) -> Result<()> {
+fn connection_listener(addr: SocketAddr, conn_tx: Sender<(TcpStream, String)>) -> Result<()> {
     let listener = TcpListener::bind(addr)?;
     loop {
-        conn_tx.send(listener.accept()?).unwrap();
+        let (mut stream, addr) = listener.accept()?;
+        let Ok(req) = deserialize::<_, ConnectionRequest>(&mut stream) else {
+            log::warn!("Failed connection from {}; bad request", addr);
+            continue;
+        };
+
+        conn_tx.send((stream, req.username)).unwrap();
     }
 }
 
+/// A single tracked connection
 struct Connection {
+    /// TCP stream
     stream: TcpStream,
-    addr: SocketAddr,
+    // /// Address
+    // addr: SocketAddr,
+    /// Message read buffer
     msg_buf: AsyncBufferedReceiver,
+    /// Connection ID
     id: ClientId,
+    /// Username
+    username: String,
 }
 
+/// Server internals
 struct Server {
+    /// ChatImproVR engine
     engine: Engine,
-    conn_rx: Receiver<(TcpStream, SocketAddr)>,
+    /// Incoming connections (Socket, Username)
+    conn_rx: Receiver<(TcpStream, String)>,
+    /// Existing connections
     conns: Vec<Connection>,
+    /// Code hotloading
     hotload: Hotloader,
```

Comment:
Do we have anything talking about hotloading? Or is this assumed as common knowledge? (I'm saying that because idk what hotloading means in this context lol

Reply:
Ah yeah hotloading just means that cimvr will reload your plugin whenever it's wasm file changes, without having to restart the whole engine